### PR TITLE
Modify the description of the HTML filtering rules

### PR DESCRIPTION
**Pull request by ngorskikh**

### CODE REVIEW

**Review by sxgunchenko — APPROVED** [review]

File: docs/general/ad-filtering/create-own-filters.md [hunk 1]
```diff
@@ -3081,11 +3093,13 @@ Following rule will delete all `script` elements with a `banner` substring in th
...
 ```
 
-**Nested elements**
+#### `wildcard`
 
-If we are dealing with multiple nested elements and they all fall within the same HTML filtering rule, they all are going to be deleted.
+:::caution Deprecation notice
 
-* **`wildcard`**
+The use of this special attribute is deprecated in favor of the `:contains()` pseudo-class.
```
Comment (slavaleleka) [inline docs/general/ad-filtering/create-own-filters.md:3100]: but it is not deprecated for the extension yet

Comment (ngorskikh) [reply docs/general/ad-filtering/create-own-filters.md:3100]: @ameshkov What if we remove the deprecation notices until all products implement the new syntax?

Comment (Alex-302) [reply docs/general/ad-filtering/create-own-filters.md:3100]: The version numbers where support will be discontinued can be specified.

Comment (ngorskikh) [reply docs/general/ad-filtering/create-own-filters.md]: I've reworded these notices.

File: docs/general/ad-filtering/create-own-filters.md [hunk 2]
```diff
@@ -3097,7 +3111,7 @@ For example:
 
 It will check, if the code of element contains two consecutive substrings `banner` and `text`.
 
-* **`max-length`**
+#### `max-length`
```
Comment (ameshkov) [inline docs/general/ad-filtering/create-own-filters.md:3114]: @ngorskikh we should add a deprecation notice here as well.

These custom attributes violate the CSS semantics and we'd better get rid of them.

File: docs/general/ad-filtering/create-own-filters.md [hunk 3]
```diff
@@ -3111,7 +3125,7 @@ $$div[tag-content="banner"][max-length="400"]
 ```
 This rule will remove all the `div` elements, whose code contains the substring `banner` and the length of which does not exceed `400` characters.
 
-* **`min-length`**
+#### `min-length`
```
Comment (ameshkov) [inline docs/general/ad-filtering/create-own-filters.md:3128]: Also need a deprecation notice here

File: docs/general/ad-filtering/create-own-filters.md [hunk 4]
```diff
@@ -3122,7 +3136,36 @@ $$div[tag-content="banner"][min-length="400"]
...
+
+:::info Compatibility
+
+The `:contains()` pseudo-class is supported by AdGuard for Windows, Mac, and Android, **running CoreLibs version 1.13 or later**.
+
+:::
+
+Requires that the element's inner HTML contains the specified text or matches the specified regular expression.
+
+As with the `tag-content` and `wildcard` attributes, specifying a `:contains()` pseudo-class on a non-leaf selector is not supported.
```
Comment (ameshkov) [inline docs/general/ad-filtering/create-own-filters.md:3166]: Could you please add a warning admonition here? This one's a pretty serious limitation.

**Review by ameshkov — CHANGES_REQUESTED** [review]

File: docs/general/ad-filtering/create-own-filters.md [hunk 5]
```diff
@@ -3122,7 +3136,36 @@ $$div[tag-content="banner"][min-length="400"]
...
+:contains(unquoted text)
+```
+or
+```
+:contains(/reg(ular )?ex(pression)?/)
+```
+
+:::note Compatibility
+
+`:-abp-contains()` and `:has-text()` are synonyms for `:contains()`.
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:3154]: why does this string have a Compatibility admonition?

File: docs/general/ad-filtering/create-own-filters.md [hunk 6]
```diff
@@ -3122,7 +3136,36 @@ $$div[tag-content="banner"][min-length="400"]
...
+
+:::
+
+:::info Compatibility
+
+The `:contains()` pseudo-class is supported by AdGuard for Windows, Mac, and Android, **running CoreLibs version 1.13 or later**.
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:3160]: here it's more appopriate

File: docs/general/ad-filtering/create-own-filters.md [hunk 7]
```diff
@@ -3122,7 +3136,36 @@ $$div[tag-content="banner"][min-length="400"]
...
+
+:::
+
+Requires that the element's inner HTML contains the specified text or matches the specified regular expression.
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:3164]:
```suggestion
Requires that the inner HTML of the element contains the specified text or matches the specified regular expression.
```

**Review by el-termikael — CHANGES_REQUESTED** [review]

File: docs/general/ad-filtering/create-own-filters.md (hunk 5 above)
Comment (ngorskikh) [reply docs/general/ad-filtering/create-own-filters.md:3154]: It's about compatibility with other ad blockers' syntax. What is the correct title for this kind of card?

**Review by el-termikael — APPROVED** [review]

**Review by ameshkov — APPROVED** [review]

**Pull request merged**